functions/devices.js

```diff
@@ -0,0 +1,379 @@
+
+  static getState(item) {
+    return {
+      isLocked: item.state === 'ON'
+    };
+  }
+
+  static get traits() {
+    return [
+      'action.devices.traits.LockUnlock'
+    ];
+  }
+}
+
+
+class SimpleLight extends GenericDevice {
+  static get type() {
+    return 'action.devices.types.LIGHT';
+  }
+
+
+  static get tag() {
+    return 'Lighting';
+  }
+
+  static appliesTo(item) {
+    return this.hasTag(item, this.tag) && (item.type === 'Switch' || (item.type === 'Group' && item.groupType && item.groupType === 'Switch'));
+  }
+
+  static getState(item) {
+    return {
+      on: item.state === 'ON'
+    };
+  }
+
+  static get traits() {
+    return [
+      'action.devices.traits.OnOff'
+    ];
+  }
+}
+
+class DimmableLight extends GenericDevice {
+  static get type() {
+    return 'action.devices.types.LIGHT';
+  }
+
+  static get tag() {
+    return 'Lighting';
+  }
+
+  static appliesTo(item) {
+    return this.hasTag(item, this.tag) && (item.type === 'Dimmer' || (item.type === 'Group' && item.groupType && item.groupType === 'Dimmer'));
+  }
+
+  static getState(item) {
+    return {
+      on: item.state > 0,
+      brightness: item.state
+    };
+  }
+
+  static get traits() {
+    return [
+      'action.devices.traits.OnOff',
+      'action.devices.traits.Brightness'
+    ];
+  }
+}
+
+class ColorLight extends GenericDevice {
+  static get type() {
+    return 'action.devices.types.LIGHT';
+  }
+
+
+  static get tag() {
+    return 'Lighting';
+  }
+  static appliesTo(item) {
+    return this.hasTag(item, this.tag) && (item.type === 'Color' || (item.type === 'Group' && item.groupType && item.groupType === 'Color'));
+  }
+
+  static getState(item) {
+    const hsvArray = item.state.split(",").map((val) => Number(val));
+    return {
+      on: hsvArray[2] > 0,
+      brightness: hsvArray[2],
+      color: {
+        spectrumHSV: {
+          hue: hsvArray[0],
+          saturation: hsvArray[1] / 100,
+          value: hsvArray[2] / 100
+        }
+      }
+    };
+  }
+
+  static get traits() {
+    return [
+      'action.devices.traits.OnOff',
+      'action.devices.traits.Brightness',
+      'action.devices.traits.ColorSetting'
+    ];
+  }
+
+  static getAttributes(item) {
+    return {
+      colorModel: 'hsv'
+    };
+  }
+}
+
+class Blinds extends GenericDevice {
+  static get type() {
+    return 'action.devices.types.BLINDS';
+  }
+
+  static get tag() {
+    return 'Blinds';
+  }
+
+  static appliesTo(item) {
+    return this.hasTag(item, this.tag) && (item.type === 'Rollershutter' || (item.type === 'Group' && item.groupType && item.groupType === 'Rollershutter'));
+  }
+
+  static getState(item) {
+    return {
+      openPercent: 100 - Number(item.state)
+    };
+  }
+
+  static get traits() {
+    return [
+      'action.devices.traits.OpenClose',
+      'action.devices.traits.StartStop'
+    ];
+  }
+
+  static getAttributes(item) {
+    return {
+      openDirection: this.hasTag(item, 'Up') ? ['UP'] : ['DOWN']
```

Comment:
There is an error in the line.
Correct is "openDirection = ['UP', 'DOWN']; " or  "openDirection = ['LEFT', 'RIGHT'];" or "openDirection = ['IN', 'OUT'];"
Look here: https://developers.google.com/assistant/smarthome/traits/openclose

Reply:
Only required when more than one direction is supported.
So I would think we can remove that attribute completely, since openHAB does not support multiple directions.

Reply:
I was hoping to get my Hunter douglas top-down bottom-up working with these either with this binding of with extending the current hdpowerview binding. 

But you state that "openHAB does not support multiple directions." so this is a non-starter either way?

Reply:
openHAB only knows one item type for this: Rollershutter.
This type only supports the commands: UpDown, StopMove & Percent.
There is currently no way of adding a direction information.
Can you control both parts separately?

Reply:
Yes thats correct. Like visualized in the link above.

Reply:
Could you then just create a group with both parts as members?

Reply:
Yes i will try this.  but i want to combine this with upgrading to recenrt 2.5M and hopefully a new version of this binding ;-)